front/src/routes/integration/all/withings/device-page/actions.js

```diff
@@ -0,0 +1,90 @@
+import update from 'immutability-helper';
+import { RequestStatus } from '../../../../../utils/consts';
+
+const actions = store => ({
+  updateDeviceProperty(state, index, property, value) {
+    const newState = update(state, {
+      withingsDevices: {
+        [index]: {
+          [property]: {
+            $set: value
+          }
+        }
+      }
+    });
+    store.setState(newState);
+  },
+  async saveDevice(state, device) {
+    await state.httpClient.post('/api/v1/device', device);
+  },
+  async getHouses(state) {
+    store.setState({
+      withingsGetStatus: RequestStatus.Getting
+    });
+    try {
+      const params = {
+        expand: 'rooms'
+      };
+      const houses = await state.httpClient.get(`/api/v1/house`, params);
+      store.setState({
+        houses,
+        withingsGetStatus: RequestStatus.Success
+      });
+    } catch (e) {
+      store.setState({
+        withingsGetStatus: RequestStatus.Error
+      });
+    }
+  },
+  async getWithingsDevice(state) {
+    store.setState({
+      withingsGetStatus: RequestStatus.Getting
+    });
+    try {
+      const options = {};
+      const withingsDevicesReceived = await state.httpClient.get('/api/v1/service/withings/device', options);
+
+      // Build map of device image by device name
+      const mapOfDeviceImgByName = new Map();
```

Comment:
I think we prefer object instead of map?

Reply:
For more, this can be extract as constant.

Reply:
I'm not sure to understand, 
you talk about a list of object with 2 attributes(device name / img name) ? If I do that I need make a function to search element in the list and a map sounds more performant for that, no ? (I'm not a good js dev so maybe better solution exist)